tests/supervisor_test/test_supervisor.py

```diff
@@ -0,0 +1,195 @@
+        out = shell.run_check(f"docker container inspect -f '{{{{.State.Status}}}}' {container_name} || true")
+        return "running" in out
+
+    for _ in range(20):
+        if check_container_running("homeassistant") and check_container_running("hassio_supervisor"):
+            break
+
+        sleep(5)
+
+    supervisor_ip = "\n".join(
+        shell.run_check("docker inspect --format='{{.NetworkSettings.IPAddress}}' hassio_supervisor")
+    )
+
+    for _ in range(20):
+        try:
+            if shell_json(f"curl -sSL http://{supervisor_ip}/supervisor/ping").get("result") == "ok":
+                break
+        except ExecutionError:
+            pass  # avoid failure when the container is restarting
+        sleep(5)
+    else:
+        raise AssertionError("Supervisor did not start in time")
+
+
+@pytest.mark.dependency(depends=["test_start_supervisor"])
+def test_check_supervisor(shell_json):
+    # check supervisor info
+    supervisor_info = shell_json("ha supervisor info --no-progress --raw-json")
+    assert supervisor_info.get("result") == "ok", "supervisor info failed"
+    logger.info("Supervisor info: %s", supervisor_info)
+    # check network info
+    network_info = shell_json("ha network info --no-progress --raw-json")
+    assert network_info.get("result") == "ok", "network info failed"
+    logger.info("Network info: %s", network_info)
+
+
+@pytest.mark.dependency(depends=["test_check_supervisor"])
+def test_update_supervisor(shell_json):
+    supervisor_info = shell_json("ha supervisor info --no-progress --raw-json")
+    supervisor_version = supervisor_info.get("data").get("version")
+    if supervisor_version == supervisor_info.get("data").get("version_latest"):
+        logger.info("Supervisor is already up to date")
+        pytest.skip("Supervisor is already up to date")
+    else:
+        result = shell_json("ha supervisor update --no-progress --raw-json")
+        if result.get("result") == "error" and "Another job is running" in result.get("message"):
+            pass
+        else:
+            assert result.get("result") == "ok", f"Supervisor update failed: {result}"
+
+        for _ in range(40):
+            try:
+                supervisor_info = shell_json("ha supervisor info --no-progress --raw-json", timeout=90)
+                data = supervisor_info.get("data")
+                if data and data.get("version") == data.get("version_latest"):
+                    logger.info(
+                        "Supervisor updated from %s to %s: %s",
+                        supervisor_version,
+                        data.get("version"),
+                        supervisor_info,
+                    )
+                    break
+            except ExecutionError:
+                pass  # avoid failure when the container is restarting
+            sleep(5)
+        else:
+            raise AssertionError("Supervisor did not update in time")
+
+
+@pytest.mark.dependency(depends=["test_check_supervisor"])
```

Comment:
Shouldn't this be

```suggestion
@pytest.mark.dependency(depends=["test_update_supervisor"])
```

Reply:
Right, there would be subtle difference - theoretically `test_update_supervisor` could fail due to a timeout yet this test would pass. So it's better to have use this deterministic non-racy dependency :)